### ISSUE TITLE
lint-scss loops when a scss file changes

### ISSUE THREAD

**Issue by poloeater** [Nykredit/gript#14]
I have experienced the following issue in v0.0.12 and v0.0.18:
When I make a modification to app.scss in the app folder the gulp-task lint-scss starts and causes a infinite loop, where the task keeps restarting.

You can see the log of this:
[12:26:37] Starting 'lint-scss'...
[12:26:38] 7 issues found in C:\projects\xp2-client\app\app.scss
[12:26:38] app.scss:1 [W] StringQuotes: Prefer single quoted strings
[12:26:38] app.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:38] app.scss:5 [W] EmptyLineBetweenBlocks: Rule declaration should be followed by an empty line
[12:26:38] app.scss:7 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:38] app.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:38] app.scss:9 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:38] app.scss:10 [W] FinalNewline: Files should end with a trailing newline
[12:26:38] 11 issues found in C:\projects\xp2-client\app\sections\bp-text\bp-text.scss
[12:26:38] sections\bp-text\bp-text.scss:2 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:38] sections\bp-text\bp-text.scss:2 [W] PropertySortOrder: Properties should be ordered background, border-radius, padding
[12:26:38] sections\bp-text\bp-text.scss:3 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:38] sections\bp-text\bp-text.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:38] sections\bp-text\bp-text.scss:4 [W] ColorVariable: Color literals like `#f0f0f0` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:38] sections\bp-text\bp-text.scss:7 [W] MergeableSelector: Merge rule `.bptextvariable .invalid` with rule on line 1
[12:26:38] sections\bp-text\bp-text.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:38] sections\bp-text\bp-text.scss:8 [W] ImportantRule: !important should not be used
[12:26:38] sections\bp-text\bp-text.scss:8 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
[12:26:38] sections\bp-text\bp-text.scss:8 [W] ColorVariable: Color literals like `red` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:38] sections\bp-text\bp-text.scss:9 [W] FinalNewline: Files should end with a trailing newline
[12:26:38] Finished 'lint-scss' after 956 ms
[12:26:38] Starting 'styles'...
[12:26:38] Starting 'lint-scss'...
[12:26:38] Finished 'styles' after 90 ms
[12:26:38] Starting 'inject-styles'...
[12:26:39] gulp-inject 1 files into index.html.
[12:26:39] Finished 'inject-styles' after 87 ms
[12:26:39] 7 issues found in C:\projects\xp2-client\app\app.scss
[12:26:39] app.scss:1 [W] StringQuotes: Prefer single quoted strings
[12:26:39] app.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:39] app.scss:5 [W] EmptyLineBetweenBlocks: Rule declaration should be followed by an empty line
[12:26:39] app.scss:7 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:39] app.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:39] app.scss:9 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:39] app.scss:10 [W] FinalNewline: Files should end with a trailing newline
[12:26:39] 11 issues found in C:\projects\xp2-client\app\sections\bp-text\bp-text.scss
[12:26:39] sections\bp-text\bp-text.scss:2 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:39] sections\bp-text\bp-text.scss:2 [W] PropertySortOrder: Properties should be ordered background, border-radius, padding
[12:26:39] sections\bp-text\bp-text.scss:3 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:39] sections\bp-text\bp-text.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:39] sections\bp-text\bp-text.scss:4 [W] ColorVariable: Color literals like `#f0f0f0` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:39] sections\bp-text\bp-text.scss:7 [W] MergeableSelector: Merge rule `.bptextvariable .invalid` with rule on line 1
[12:26:39] sections\bp-text\bp-text.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:39] sections\bp-text\bp-text.scss:8 [W] ImportantRule: !important should not be used
[12:26:39] sections\bp-text\bp-text.scss:8 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
[12:26:39] sections\bp-text\bp-text.scss:8 [W] ColorVariable: Color literals like `red` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:39] sections\bp-text\bp-text.scss:9 [W] FinalNewline: Files should end with a trailing newline
[12:26:39] Finished 'lint-scss' after 986 ms
[12:26:39] Starting 'styles'...
[12:26:39] Starting 'lint-scss'...
[12:26:40] Finished 'styles' after 74 ms
[12:26:40] Starting 'inject-styles'...
[12:26:40] gulp-inject 1 files into index.html.
[12:26:40] Finished 'inject-styles' after 63 ms
[12:26:40] 7 issues found in C:\projects\xp2-client\app\app.scss
[12:26:40] app.scss:1 [W] StringQuotes: Prefer single quoted strings
[12:26:40] app.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:40] app.scss:5 [W] EmptyLineBetweenBlocks: Rule declaration should be followed by an empty line
[12:26:40] app.scss:7 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:40] app.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:40] app.scss:9 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:40] app.scss:10 [W] FinalNewline: Files should end with a trailing newline
[12:26:40] 11 issues found in C:\projects\xp2-client\app\sections\bp-text\bp-text.scss
[12:26:40] sections\bp-text\bp-text.scss:2 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:40] sections\bp-text\bp-text.scss:2 [W] PropertySortOrder: Properties should be ordered background, border-radius, padding
[12:26:40] sections\bp-text\bp-text.scss:3 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:40] sections\bp-text\bp-text.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:40] sections\bp-text\bp-text.scss:4 [W] ColorVariable: Color literals like `#f0f0f0` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:40] sections\bp-text\bp-text.scss:7 [W] MergeableSelector: Merge rule `.bptextvariable .invalid` with rule on line 1
[12:26:40] sections\bp-text\bp-text.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:40] sections\bp-text\bp-text.scss:8 [W] ImportantRule: !important should not be used
[12:26:40] sections\bp-text\bp-text.scss:8 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
[12:26:40] sections\bp-text\bp-text.scss:8 [W] ColorVariable: Color literals like `red` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:40] sections\bp-text\bp-text.scss:9 [W] FinalNewline: Files should end with a trailing newline
[12:26:40] Finished 'lint-scss' after 987 ms
[12:26:40] Starting 'styles'...
[12:26:40] Starting 'lint-scss'...
[12:26:41] Finished 'styles' after 38 ms
[12:26:41] Starting 'inject-styles'...
[12:26:41] gulp-inject 1 files into index.html.
[12:26:41] Finished 'inject-styles' after 54 ms
[12:26:41] 7 issues found in C:\projects\xp2-client\app\app.scss
[12:26:41] app.scss:1 [W] StringQuotes: Prefer single quoted strings
[12:26:41] app.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:41] app.scss:5 [W] EmptyLineBetweenBlocks: Rule declaration should be followed by an empty line
[12:26:41] app.scss:7 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:41] app.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:41] app.scss:9 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:41] app.scss:10 [W] FinalNewline: Files should end with a trailing newline
[12:26:41] 11 issues found in C:\projects\xp2-client\app\sections\bp-text\bp-text.scss
[12:26:41] sections\bp-text\bp-text.scss:2 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:41] sections\bp-text\bp-text.scss:2 [W] PropertySortOrder: Properties should be ordered background, border-radius, padding
[12:26:41] sections\bp-text\bp-text.scss:3 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:41] sections\bp-text\bp-text.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:41] sections\bp-text\bp-text.scss:4 [W] ColorVariable: Color literals like `#f0f0f0` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:41] sections\bp-text\bp-text.scss:7 [W] MergeableSelector: Merge rule `.bptextvariable .invalid` with rule on line 1
[12:26:41] sections\bp-text\bp-text.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:41] sections\bp-text\bp-text.scss:8 [W] ImportantRule: !important should not be used
[12:26:41] sections\bp-text\bp-text.scss:8 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
[12:26:41] sections\bp-text\bp-text.scss:8 [W] ColorVariable: Color literals like `red` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:41] sections\bp-text\bp-text.scss:9 [W] FinalNewline: Files should end with a trailing newline
[12:26:41] Finished 'lint-scss' after 935 ms
[12:26:41] Starting 'styles'...
[12:26:41] Starting 'lint-scss'...
[12:26:41] Finished 'styles' after 45 ms
[12:26:41] Starting 'inject-styles'...
[12:26:41] gulp-inject 1 files into index.html.
[12:26:42] Finished 'inject-styles' after 50 ms
[12:26:42] 7 issues found in C:\projects\xp2-client\app\app.scss
[12:26:42] app.scss:1 [W] StringQuotes: Prefer single quoted strings
[12:26:42] app.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:42] app.scss:5 [W] EmptyLineBetweenBlocks: Rule declaration should be followed by an empty line
[12:26:42] app.scss:7 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:42] app.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:42] app.scss:9 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:42] app.scss:10 [W] FinalNewline: Files should end with a trailing newline
[12:26:42] 11 issues found in C:\projects\xp2-client\app\sections\bp-text\bp-text.scss
[12:26:42] sections\bp-text\bp-text.scss:2 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:42] sections\bp-text\bp-text.scss:2 [W] PropertySortOrder: Properties should be ordered background, border-radius, padding
[12:26:42] sections\bp-text\bp-text.scss:3 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:42] sections\bp-text\bp-text.scss:4 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:42] sections\bp-text\bp-text.scss:4 [W] ColorVariable: Color literals like `#f0f0f0` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:42] sections\bp-text\bp-text.scss:7 [W] MergeableSelector: Merge rule `.bptextvariable .invalid` with rule on line 1
[12:26:42] sections\bp-text\bp-text.scss:8 [W] Indentation: Line should be indented 2 spaces, but was indented 4 spaces
[12:26:42] sections\bp-text\bp-text.scss:8 [W] ImportantRule: !important should not be used
[12:26:42] sections\bp-text\bp-text.scss:8 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
[12:26:42] sections\bp-text\bp-text.scss:8 [W] ColorVariable: Color literals like `red` should only be used in variable declarations; they should be referred to via variable everywhere else.
[12:26:42] sections\bp-text\bp-text.scss:9 [W] FinalNewline: Files should end with a trailing newline
[12:26:42] Finished 'lint-scss' after 946 ms
[12:26:42] Starting 'styles'...
[12:26:42] Starting 'lint-scss'...
[12:26:42] Finished 'styles' after 54 ms

**Comment by jotka** [comment]
Hi, @poloeater. This is something I cannot reproduce. Can you please change the node_modules/gript/tasks/watch.gulp.js in your project into [this gist](https://gist.github.com/jotka/1b60afb72926fd9f80df#file-watch-gulp-js) and report back ? 
This will give gulp-watch additional options:

```js
watchOptions = {
            usePolling: true,
            awaitWriteFinish: {
                stabilityThreshold: 2000,
                pollInterval: 100
            }
        };
```

**Comment by poloeater** [comment]
Hi @jotka 

I tried using your watch.gulp.js and then it stopped happening. Going back to the original watch.gulp.js triggers the bug again. So I can't give you a log with the watchoptions because it simply doesn't happen with these settings :-/

**Comment by jotka** [comment]
ok, this will probably go to the npm release 0.0.20

**Comment by jotka** [comment]
fixed and published in npm 0.0.20, closing.
